src/Discord.Net.Commands/CommandService.cs

```diff
@@ -139,15 +142,31 @@ public CommandService(CommandServiceConfig config)
                 _moduleLock.Release();
             }
         }
-        private ModuleInfo LoadModuleInternal(ModuleInfo module)
+        private ModuleInfo LoadModuleInternal(ModuleInfo module, IServiceProvider services)
```

Comment:
Considering that the signature is now OnModuleBuilding, wouldn't it be better to call it in ModuleBuilder.Build instead?

Reply:
That's a *ton* more stuff to pass the `IServiceProvider` to.
```diff
#  ModuleBuilder.cs
- private ModuleInfo BuildImpl(CommandService service, ModuleInfo parent = null);
+ private ModuleInfo BuildImpl(CommandService service, IServiceProvider services, ModuleInfo parent = null);

- public ModuleInfo Build(CommandService service);
+ public ModuleInfo Build(CommandService service, IServiceProvider services);
 
- internal ModuleInfo Build(CommandService service, ModuleInfo parent);
+ internal ModuleInfo Build(CommandService service, IServiceProvider services, ModuleInfo parent);

# ModuleClassBuilder.cs
- public static Task<Dictionary<Type, ModuleInfo>> BuildAsync(CommandService service, params TypeInfo[] validTypes);
+ public static Task<Dictionary<Type, ModuleInfo>> BuildAsync(CommandService service, IServiceProvider services, params TypeInfo[] validTypes);

- public static async Task<Dictionary<Type, ModuleInfo>> BuildAsync(IEnumerable<TypeInfo> validTypes, CommandService service);
+ public static async Task<Dictionary<Type, ModuleInfo>> BuildAsync(IEnumerable<TypeInfo> validTypes, CommandService service, IServiceProvider services)

# CommandService.cs
- public async Task<ModuleInfo> CreateModuleAsync(string primaryAlias, Action<ModuleBuilder> buildFunc);
+ public async Task<ModuleInfo> CreateModuleAsync(string primaryAlias, Action<ModuleBuilder> buildFunc, IServiceProvider services);

# ModuleInfo.cs
- internal ModuleInfo(ModuleBuilder builder, CommandService service, ModuleInfo parent = null);
+ internal ModuleInfo(ModuleBuilder builder, CommandService service, IServiceProvider services, ModuleInfo parent = null);

- private List<ModuleInfo> BuildSubmodules(ModuleBuilder parent, CommandService service);
+ private List<ModuleInfo> BuildSubmodules(ModuleBuilder parent, CommandService service, IServiceProvider services);
```

Reply:
Honestly, why aren't we asking users to set the `IServiceProvider` with the `CommandServiceConfig`? Then wherever the `CommandService` flows, we could access it, and we wouldn't have to change *any* signatures like this. 🤔

Reply:
Passing the services in CommandServiceConfig would not allow scoped services to work, unless we forcibly ran commands in a new scope internally.

Reply:
Is that the only place where scoped dependencies could be an issue? Because that doesn't sound *that* bad of a change.